src/instant.rs

```diff
@@ -173,6 +173,59 @@ macro_rules! impl_instant_for_integer {
                     }
                 }
             }
+
+            /// Try to convert to an [`Instant`] with a different fraction.
+            ///
+            /// Should optimize down to one multiply and one divide.
+            ///
+            /// ```
+            /// # use fugit::*;
+            #[doc = concat!("let i = Instant::<", stringify!($i), ", 1, 1_000>::from_ticks(40);")]
+            /// let d = i.try_into::<1, 250>().unwrap();
+            /// assert_eq!(d.ticks(), 10);
+            ///
+            #[doc = concat!("let i = Instant::<", stringify!($i), ", 1, 1_000>::from_ticks(", stringify!($i), "::MAX - 10);")]
+            /// let d = i.try_into::<7, 2_000>();
+            /// assert_eq!(d, None);
+            /// ```
+            pub const fn try_into<const O_NOM: u32, const O_DENOM: u32>(
+                self,
+            ) -> Option<Instant<$i, O_NOM, O_DENOM>> {
+                if let Some(ticks) = self
+                    .ticks
+                    .checked_mul(Helpers::<NOM, DENOM, O_NOM, O_DENOM>::RD_TIMES_LN as $i)
+                {
+                    Some(Instant::<$i, O_NOM, O_DENOM>::from_ticks(
+                        ticks / Helpers::<NOM, DENOM, O_NOM, O_DENOM>::LD_TIMES_RN as $i,
+                    ))
+                } else {
+                    None
+                }
+            }
+
+            /// Infallibly convert to an [`Instant`] with a different fraction.
+            ///
+            /// If the user provides incompatible fractions (i.e., to do the conversion we must multiply by something
+            /// other than one), we will fail to compile, even if the conversion wouldn't have overflown:
+            /// ```compile_fail
+            #[doc = concat!("let i = Instant::<", stringify!($i), ", 1, 1_000>::from_ticks(42);")]
+            /// let d = i.into::<7, 2_000>();
+            /// ```
+            pub fn into<const O_NOM: u32, const O_DENOM: u32>(
```

Comment:
realizing that I accidentally left this as non-`const` and now it's not actually failing to compile...

Reply:
I can't seem to get this to work. I can only get runtime assert to work. If I do:
```rust
const EQ: usize = (Helpers::<NOM, DENOM, O_NOM, O_DENOM>::RD_TIMES_LN as usize - 1 as usize) + (1 as usize - Helpers::<NOM, DENOM, O_NOM, O_DENOM>::RD_TIMES_LN as usize);
```

I get a bunch of fun errors:
```rust
error[E0401]: can't use generic parameters from outer function
   --> src/instant.rs:220:155
    |
215 |             pub const fn into<const O_NOM: u32, const O_DENOM: u32>(
    |                                                       ------- const parameter from outer function
...
220 |                 const EQ: usize = (Helpers::<NOM, DENOM, O_NOM, O_DENOM>::RD_TIMES_LN as usize - 1 as usize) + (1 as usize - Helpers::<NOM, DENOM, O_NOM, O_DENOM>::...
    |                                                                                                                                                           ^^^^^^^ use of generic parameter from outer function
```

How do you feel about a `debug_assert!()` instead?